src/main/java/com/rsr5/gocd/riemann_notifier/GoNotificationPlugin.java

```diff
@@ -138,23 +144,68 @@ private GoPluginApiResponse handleNotificationsInterestedIn() {
         return renderJSON(SUCCESS_RESPONSE_CODE, response);
     }
 
-    private String service(JsonObject json) {
-        JsonObject pipelineObject, stageObject;
-        pipelineObject = (JsonObject) json.get("pipeline");
-        stageObject = (JsonObject) pipelineObject.get("stage");
+    private String safeGetAsString(JsonObject json, String property) {
+        JsonElement element = json.get(property);
+        if (element != null && element.isJsonPrimitive()) {
+            return element.getAsString();
+        }
+        return null;
+    }
+
+    final class StageData {
+        public String pipeline;
+        public String stage;
+        public String state;
+        public String counter;
+        public Instant start;
+        public Instant end;
+        public List<JobData> jobs;
+
+        final class JobData {
+            public String name;
+            public String result;
+            public Instant start;
+            public Instant end;
+
+            public JobData(JsonObject json) {
+                this.name = safeGetAsString(json, "name");
+                this.result = safeGetAsString(json, "result");
+                this.start = Instant.parse(safeGetAsString(json, "schedule-time"));
+                this.end = Instant.parse(safeGetAsString(json, "complete-time"));
```

Comment:
Unsure if this would be impacting, but `Instant.parse(...)` will throw if the argument is `null`. It looks like `safeGetAsString` potentially returns `null` if the JSON property we're looking for is missing or is a JSON primitive.